pkg/tsdb/postgres/postgres.go

```diff
@@ -56,15 +57,117 @@ func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndp
 	return endpoint, err
 }
 
+func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
+	logger := log.New("tsdb.postgres")
+	logger.Debug("Creating Postgres query endpoint")
+	endpoint, err := databaseConnection(datasource, logger)
+	return endpoint, err
+}
+
 // escape single quotes and backslashes in Postgres connection string parameters.
 func escape(input string) string {
 	return strings.ReplaceAll(strings.ReplaceAll(input, `\`, `\\`), "'", `\'`)
 }
 
+func writeConnectionFile(
+	ds *models.DataSource, fileContent string, currentPath string, certFileName string, jsonFieldName string) error {
+	var generatedFilePath string
+	if ssljs, ok := ds.JsonData.CheckGet(jsonFieldName); ok {
+		generatedFilePath = ssljs.MustString("")
+	}
+	if len(fileContent) > 0 {
+		if len(generatedFilePath) == 0 {
+			generatedFilePath = filepath.Join(currentPath, certFileName)
+		}
+
+		if err := ioutil.WriteFile(generatedFilePath, []byte(fileContent), 0600); err != nil {
+			return err
+		}
+		ds.JsonData.Set(jsonFieldName, generatedFilePath)
+	} else {
+		if len(generatedFilePath) > 0 {
+			if err := os.Remove(generatedFilePath); err != nil {
+				return err
+			}
+		}
+		ds.JsonData.Set(jsonFieldName, "")
+	}
+	return nil
+}
+
+func writeConnectionFiles(ds *models.DataSource, logger log.Logger) error {
+	sslMode := strings.TrimSpace(strings.ToLower(ds.JsonData.Get("sslmode").MustString("verify-full")))
+	decrypted := ds.SecureJsonData.Decrypt()
+	tlsCACert := decrypted["tlsCACert"]
+	tlsClientCert := decrypted["tlsClientCert"]
+	tlsClientKey := decrypted["tlsClientKey"]
+
+	if sslMode != "disable" {
```

Comment:
Better checking if the `sslMode` is disabled and then return directly in order to reduce the nested block.